presentation-compiler/src/main/dotty/tools/pc/completions/Completions.scala

```diff
@@ -69,113 +78,40 @@ class Completions(
+        hasSyntheticCursorSuffix
+
+    def isWildcardParam(sym: Symbol) =
+      if sym.isTerm && sym.owner.isAnonymousFunction then
+        sym.name match
+          case DerivedName(under, _) =>
+            under.isEmpty
           case _ => false
-      end if
-    end include
+      else false
 
-    def allowBracketSuffix: Boolean =
-      this match
-        case Type(hasTypeParams, _) => !hasTypeParams
-        case _ => false
-
-    def allowTemplateSuffix: Boolean =
-      this match
-        case Type(_, hasNewKw) => hasNewKw
-        case _ => false
-
-    def allowApplicationSuffix: Boolean =
-      this match
-        case Term => true
-        case _ => false
-
-  end CursorPos
-
-  private lazy val cursorPos =
-    calculateTypeInstanceAndNewPositions(path)
-
-  private def calculateTypeInstanceAndNewPositions(
-      path: List[Tree]
-  ): CursorPos =
-    path match
-      case (_: Import) :: _ => CursorPos.Import
-      case _ :: (_: Import) :: _ => CursorPos.Import
-      case (head: (Select | Ident)) :: tail =>
-        // https://github.com/lampepfl/dotty/issues/15750
-        // due to this issue in dotty, because of which trees after typer lose information,
-        // we have to calculate hasNoSquareBracket manually:
-        val hasSquareBracket =
-          val span: Span = head.srcPos.span
-          if span.exists then
-            var i = span.end
-            while i < (text.length() - 1) && text(i).isWhitespace do i = i + 1
-
-            if i < text.length() then text(i) == '['
-            else false
-          else false
-
-        def typePos = CursorPos.Type(hasSquareBracket, hasNewKw = false)
-        def newTypePos =
-          CursorPos.Type(hasSquareBracket, hasNewKw = true)
-
-        tail match
-          case (v: ValOrDefDef) :: _ if v.tpt.sourcePos.contains(pos) =>
-            typePos
-          case New(selectOrIdent: (Select | Ident)) :: _
-              if selectOrIdent.sourcePos.contains(pos) =>
-            newTypePos
-          case (a @ AppliedTypeTree(_, args)) :: _
-              if args.exists(_.sourcePos.contains(pos)) =>
-            typePos
-          case (templ @ Template(constr, _, self, _)) :: _
-              if (constr :: self :: templ.parents).exists(
-                _.sourcePos.contains(pos)
-              ) =>
-            typePos
-          case _ =>
-            CursorPos.Term
-        end match
-
-      case (_: TypeTree) :: TypeApply(Select(newQualifier: New, _), _) :: _
-          if newQualifier.sourcePos.contains(pos) =>
-        CursorPos.Type(hasTypeParams = false, hasNewKw = true)
-
-      case _ => CursorPos.Term
-    end match
-  end calculateTypeInstanceAndNewPositions
+    if generalExclude then false
+    else if completionMode.is(Mode.ImportOrExport) then true
+    else if completionMode.is(Mode.Term) && isWildcardParam(sym) then false
+    else if completionMode.is(Mode.Term) && (sym.isTerm || sym.is(Package)) then true
+    else !completionMode.is(Mode.Term)
```

Comment:
The problem is you can't use pattern matching on it because `completionMode` is basically an `Int`, which is shifted accordingly to represent a Mode. I think this is the best that we can get :D

Reply:
I know, but isn't this:
```demo.scala
if generalExclude then false
else if !completionMode.is(Mode.Term) then true
else !isWildcardParam(sym) && (sym.isTerm || sym.is(Package))
```
same?

Reply:
Thanks!